router_types.c

```diff
@@ -44,6 +47,55 @@ static const struct router_type *router_types[] = {
 	NULL,
 };
 
+int read_mac(uint8_t mac[ETH_ALEN], const char *macstr)
+{
+	int ret;
+
+	ret = sscanf(macstr, "%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX",
+				 &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+	if (ret != 6)
+		ret = sscanf(macstr, "%02hhX-%02hhX-%02hhX-%02hhX-%02hhX-%02hhX",
+					 &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
+
+	return (ret == 6);
+}
+
+void mac_whitelist_add(char *macstr)
+{
+	uint8_t mac[ETH_ALEN];
+	if (read_mac(mac, macstr)) {
+		if (mac_whitelist_head == NULL) {
+			mac_whitelist_head = malloc(sizeof(struct mac_whitelist_entry));
+			mac_whitelist_head->next = NULL;
+			memcpy(mac_whitelist_head->mac, mac, ETH_ALEN);
+		}
+		else {
```

Comment:
If this would be needed at all (see next comment for this function) then I would exect the closing curly bracket on the same line as the else like this:

    } else {